src/o365/spfx/commands/project/project-rename.ts

```diff
@@ -90,81 +90,101 @@ class SpfxProjectRenameCommand extends BaseProjectCommand {
       }
     }
 
-    const filePaths: string[] = [
-      path.join(this.projectRootPath, 'package.json'),
-      path.join(this.projectRootPath, '.yo-rc.json'),
-      path.join(this.projectRootPath, 'config/package-solution.json'),
-      path.join(this.projectRootPath, 'config/deploy-azure-storage.json'),
-      path.join(this.projectRootPath, 'README.md')
-    ];
+    const replacePackageJsonContent = (filePath: string) => {
+      if (!fs.existsSync(filePath)) {
+        return;
+      }
+      const existingContent = fs.readFileSync(filePath, 'utf-8');
+      const updatedContent = JSON.parse(existingContent);
+      if (updatedContent &&
+        updatedContent.name) {
+        updatedContent.name = args.options.newName;
+      }
+      fs.writeFileSync(filePath, JSON.stringify(updatedContent, null, 2), 'utf-8');
```

Comment:
If the file contents have not been changed, we shouldn't write the file back to the disk